### PR TITLE
Add pin controller driver for W91

### PR DESCRIPTION
**Pull request by serhiiSalamakha**
http://192.168.48.49:8080/browse/ZEPHYR-428

### CODE REVIEW

File: boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-common.dtsi [hunk 1]
```diff
@@ -50,6 +50,8 @@
 	};
 
 	chosen {
+		zephyr,console = &uart1;
```
Comment (andriy-bilynskyy) [inline boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-common.dtsi]: we have special debug backend on uart0. Why do we relocate this?

File: boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-pinctrl.dtsi [hunk 2]
```diff
@@ -5,3 +5,40 @@
...
 
 #include <zephyr/dt-bindings/pinctrl/w91-pinctrl.h>
+
+&pinctrl {
+	/* Set pad-mul-sel register value.
+	 * Note: Pins functions below (pinmux = <...>) depend on this value.
+	 */
+	pad-mul-sel = <1>;
+
+	/* UART0: TX(P22), RX(P21), RTS(P20), CTS(P19) */
```
Comment (andriy-bilynskyy) [inline boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-pinctrl.dtsi]: uart0 is reserved for debugging. we shouldn't add it here

File: boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-pinctrl.dtsi [hunk 3]
```diff
@@ -5,3 +5,40 @@
...
+	};
+	uart0_rts_p20_default: uart0_rts_p20_default {
+		pinmux = <W91_PINMUX_SET(W91_PIN_20, W91_FUNC_UART0_RTS)>;
+	};
+	uart0_cts_p19_default: uart0_cts_p19_default {
+		pinmux = <W91_PINMUX_SET(W91_PIN_19, W91_FUNC_UART0_CTS)>;
+	};
+
+	/* UART1: TX(P1), RX(P0), RTS(P20), CTS(P19) */
+
```
Comment (andriy-bilynskyy) [inline boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-pinctrl.dtsi]: uart1 is used for debug in core N22...

File: boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d_defconfig [hunk 4]
```diff
@@ -7,9 +7,9 @@ CONFIG_BOARD_TLSR9118BDK40D=y
 CONFIG_GPIO=y
 CONFIG_SYS_CLOCK_TICKS_PER_SEC=1000
 CONFIG_UART_INTERRUPT_DRIVEN=n
-CONFIG_SERIAL=n
-CONFIG_CONSOLE=n
-CONFIG_UART_CONSOLE=n
+CONFIG_SERIAL=y
```
Comment (andriy-bilynskyy) [inline boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d_defconfig]: We are making conflicts with our debug backend

File: drivers/pinctrl/pinctrl_w91.c [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+		}
+
+		pins++;
+	}
+
+	return err;
+}
+
+SYS_INIT(pinctrl_w91_init, POST_KERNEL, CONFIG_TELINK_W91_IPC_DRIVERS_INIT_PRIORITY);
+SYS_INIT(pinctrl_w91_pin_map_set, POST_KERNEL, CONFIG_PINCTRL_TELINK_W91_PIN_MAP_SET_PRIORITY);
```
Comment (andriy-bilynskyy) [inline drivers/pinctrl/pinctrl_w91.c]: Seems like driver workaround not a driver.
When some device with pinctrl functionality is initing it configures its pins...

**Review by andriy-bilynskyy — CHANGES_REQUESTED** [review]

File: boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-common.dtsi (hunk 1 above)
Comment (serhiiSalamakha) [reply boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-common.dtsi]: Updated

File: boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-pinctrl.dtsi (hunk 2 above)
Comment (serhiiSalamakha) [reply boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-pinctrl.dtsi]: Updated

File: boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-pinctrl.dtsi (hunk 3 above)
Comment (serhiiSalamakha) [reply boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-pinctrl.dtsi]: Updated

File: boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d_defconfig (hunk 4 above)
Comment (serhiiSalamakha) [reply boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d_defconfig]: Updated

File: drivers/pinctrl/pinctrl_w91.c (hunk 5 above)
Comment (serhiiSalamakha) [reply drivers/pinctrl/pinctrl_w91.c]: Updated

**Review by andriy-bilynskyy — APPROVED** [review]

**Pull request merged**